app/services/void_participant_declaration.rb

```diff
@@ -45,4 +47,16 @@ def make_voided
   def line_item
     participant_declaration.statement_line_items.find_by(state: %w[eligible payable submitted])
   end
+
+  def participant_profile
+    @participant_profile ||= participant_declaration.participant_profile
+  end
+
+  def check_mentor_completion
+    Mentors::CheckTrainingCompletion.call(mentor_profile: participant_profile) if mentor_completion_event?
+  end
+
+  def mentor_completion_event?
+    participant_profile.mentor? && participant_declaration.declaration_type == "completed"
```

Comment:
might be worth moving this logic to the service as it's repeated? we can return early there

Reply:
Yes, I've moved this now into a service that performs the check and calls the service only when necessary.